saos-persistence/src/main/java/pl/edu/icm/saos/persistence/common/IndexableObject.java

```diff
@@ -5,30 +5,49 @@
 
 import org.joda.time.DateTime;
 
+/**
+ * Abstract entity class representing persistence data object
+ * that can be indexed
+ * 
+ * @author madryk
+ */
 @MappedSuperclass
 public abstract class IndexableObject extends DataObject {
 
     private boolean indexed = false;
     
     private DateTime indexedDate;
-    
-    public void markAsIndexed() {
-        indexed = true;
-        indexedDate = new DateTime();
-    }
-    
-    public void resetIndexedFlag() {
-        indexed = false;
-    }
 
+
+    /**
+     * Is entity needs to be indexed.
+     */
     @Column(columnDefinition=ColumnDefinitionConst.BOOLEAN_NOT_NULL_DEFUALT_FALSE)
     public boolean isIndexed() {
         return indexed;
     }
 
+    /**
+     * Returns date when entity was last indexed.
+     */
     public DateTime getIndexedDate() {
         return indexedDate;
     }
+    
+    
+    /**
+     * Sets indexed flag to false.
+     * <br/>
+     * Method is meant to be used only in persistence
+     * repository classes.
+     * Do not use it outside of that scope.
+     * Indexed flag will be automatically reset when
+     * saving the entity.
+     */
```

Comment:
Sets the indexed date to false.


This method is used in persistence repository classes, so that saving the object resets the indexed flag. You should not use it in other classes.